src/services/AttachmentApi.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * @copyright Copyright (c) 2020 Jakob Röhrl <jakob.roehrl@web.de>
+ *
+ * @author Jakob Röhrl <jakob.roehrl@web.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import axios from '@nextcloud/axios'
+
+export class AttachmentApi {
+
+	url(url) {
+		url = `/apps/deck${url}`
+		// eslint-disable-next-line
+		return OC.generateUrl(url)
+	}
+
+	async fetchAttachments(cardId) {
+		const response = await axios({
+			method: 'GET',
+			url: this.url(`/cards/${cardId}/attachments`),
+		})
+		return response.data
+	}
+
+	async createAttachment({ cardId, formData }) {
+		try {
+			const response = await axios({
+				method: 'POST',
+				url: this.url(`/cards/${cardId}/attachment`),
+				data: formData,
+			})
+			return response.data
+		} catch (e) {
```

Comment:
I would prefer to have this catch being thrown to the vue component that calls the vuex action, so we can show a proper warning in the frontend that the upload has failed.

Reply:
Make sense, how can I do it?

Reply:
If you want to keep the logging in the AttachmentApi just add a `throw e` after the log entry. That makes sure that the error continues to propagate out of the function.